astropy/wcs/src/util.c

```diff
@@ -20,18 +20,28 @@ void set_invalid_to_nan(
   const int* s = stat;
   const int* s_end = stat + ncoord;
   double n;
+  int bit;
 
   #ifndef NAN
     #define INF (DBL_MAX+DBL_MAX)
     #define NAN (INF-INF)
   #endif
 
+  // Note that stat is a bit mask, so we need to mask only some of
+  // the coordinates depending on the bit mask values.
+
   n = NAN;
 
   for ( ; s != s_end; ++s) {
     if (*s) {
+      bit = 1;
       for (i = 0; i < nelem; ++i) {
-        *d++ = n;
+        if ((*s & bit) == bit) {
```

Comment:
I'm more used to the `if (*s & bit)` but preferences may vary. Is there any specific reason to check for equality to bit?

Reply:
Oh yes of course, I was being silly